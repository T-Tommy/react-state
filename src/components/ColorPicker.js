import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class ColorPicker extends PureComponent {
  static propTypes = {
    colors: PropTypes.array.isRequired
  };

  state = {
    color: 'blue'
  };

  handleClick = (color) => {
    return () => {
      this.setState({ color });
    };
  };

  render() {
    const divStyle = {
      background: this.state.color,
      height: '100px',
      width: '100px'
    };

    // somehow separate button component and splotch/div component
    return (
      <>
        <section>
          {
            this.props.colors
              .map(color =>
                <button key={ color } onClick={ this.handleClick(color.toLowerCase()) }>
                  { color }
                </button>
              )
          }
        </section>
        <div style={ divStyle }></div>
      </>
    );
  }
}
