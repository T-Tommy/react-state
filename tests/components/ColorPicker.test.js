import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from '../../src/components/ColorPicker';

describe('ColorPicker component', () => {
  it('renders the ColorPicker', () => {
    const wrapper = shallow(<ColorPicker colors={ ['Red', 'Orange', 'Yellow'] } />);
    expect(wrapper).toMatchSnapshot();
  });

  it('changes color to red on button press', () => {
    const wrapper = shallow(<ColorPicker colors={ ['Red', 'Orange', 'Yellow'] } />);
    wrapper.find('button').at(0).simulate('click');
    expect(wrapper.state('color')).toEqual('red');
  });

  it('changes color to orange on button press', () => {
    const wrapper = shallow(<ColorPicker colors={ ['Red', 'Orange', 'Yellow'] } />);
    wrapper.find('button').at(1).simulate('click');
    expect(wrapper.state('color')).toEqual('orange');
  });

  it('changes color to yellow on button press', () => {
    const wrapper = shallow(<ColorPicker colors={ ['Red', 'Orange', 'Yellow'] } />);
    wrapper.find('button').at(2).simulate('click');
    expect(wrapper.state('color')).toEqual('yellow');
  });
});
