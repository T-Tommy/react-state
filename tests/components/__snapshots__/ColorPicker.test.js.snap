// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker component renders the ColorPicker 1`] = `
<Fragment>
  <section>
    <button
      key="Red"
      onClick={[Function]}
    >
      Red
    </button>
    <button
      key="Orange"
      onClick={[Function]}
    >
      Orange
    </button>
    <button
      key="Yellow"
      onClick={[Function]}
    >
      Yellow
    </button>
  </section>
  <div
    style={
      Object {
        "background": "blue",
        "height": "100px",
        "width": "100px",
      }
    }
  />
</Fragment>
`;
